### PR TITLE
common/blkdev: handle devices with ID_MODEL as "LVM PV ..." but valid ID_MODEL_ENC

### PR DESCRIPTION
**Pull request by liewegas**
Some devices appear like so:

E: ID_MODEL=LVM PV LUClYG-Oyte-jcM6-npfZ-ncsl-ycL0-bkOH0m on /dev/sdn
E: ID_MODEL_ENC=WDC\x20WDS200T2B0A-00SM50\x20\x20\x20\x20\x20\x20\x20\x20\x20\x20\x20\x20\x20\x20\x20\x20\x20\x20
E: ID_SERIAL=WDC_WDS200T2B0A-00SM50_183503800168
E: ID_SERIAL_SHORT=183503800168

To avoid upsetting our prioritization, defer to ID_MODEL_ENC over ID_MODEL
if ID_MODEL has "LVM PV" in it (as opposed to just using ID_SERIAL).

Signed-off-by: Sage Weil <sage@redhat.com>

### CODE REVIEW

File: src/common/blkdev.cc [hunk 1]
```diff
@@ -433,6 +433,23 @@ bool get_vdo_utilization(int fd, uint64_t *total, uint64_t *avail)
...
+    if (p == std::string::npos) {
+      break;
+    }
+    v = v.substr(0, p) + ' ' + v.substr(p + 4);
+  }
+  while (v.size() && v[v.size()-1] == ' ') {
+    v.resize(v.size() - 1);
+  }
+  std::replace(v.begin(), v.end(), ' ', '_');
+  return v;
```
Comment (tchaikov) [inline src/common/blkdev.cc]: nit, an alternative could be
```c++
#include <boost/algorithm/string/replace.hpp>
// ....
  auto v = boost::replace_all_copy(in, "\\x20", " ");
  if (auto found = v.find_last_not_of(" "); found != v.npos) {
    v.erase(found + 1);
  }
  std::replace(v.begin(), v.end(), ' ', '_');
  return v;
```

File: src/test/common/test_blkdev.cc [hunk 2]
```diff
@@ -91,3 +91,19 @@ TEST_F(BlockDevTest, is_rotational)
...
+  const char *foo[][2] = {
+    { "WDC\\x20WDS200T2B0A-00SM50\\x20\\x20\\x20\\x20\\x20\\x20\\x20\\x20\\x20\\x20\\x20\\x20\\x20\\x20\\x20\\x20\\x20\\x20",
+      "WDC_WDS200T2B0A-00SM50" },
+    { 0, 0},
+  };
+
+  for (unsigned i = 0; foo[i][0]; ++i) {
+    std::string d = _decode_model_enc(foo[i][0]);
+    cout << " '" << foo[i][0] << "' -> '" << d << "'" << std::endl;
+    ASSERT_EQ(d, std::string(foo[i][1]));
```
Comment (tchaikov) [inline src/test/common/test_blkdev.cc]: might want to transpose `d` and `std::string(foo[i][1])`. the first argument is expected value.

File: src/test/common/test_blkdev.cc [hunk 3]
```diff
@@ -91,3 +91,19 @@ TEST_F(BlockDevTest, is_rotational)
...
+
+  const char *foo[][2] = {
+    { "WDC\\x20WDS200T2B0A-00SM50\\x20\\x20\\x20\\x20\\x20\\x20\\x20\\x20\\x20\\x20\\x20\\x20\\x20\\x20\\x20\\x20\\x20\\x20",
+      "WDC_WDS200T2B0A-00SM50" },
+    { 0, 0},
+  };
+
+  for (unsigned i = 0; foo[i][0]; ++i) {
+    std::string d = _decode_model_enc(foo[i][0]);
+    cout << " '" << foo[i][0] << "' -> '" << d << "'" << std::endl;
```
Comment (tchaikov) [inline src/test/common/test_blkdev.cc]: i'd suggest remove this output. the `ASSERT_*` and `EXPECT_*` would be good enough for test. they will print out the value of both sides if they does not match.

File: src/test/common/test_blkdev.cc [hunk 4]
```diff
@@ -91,3 +91,19 @@ TEST_F(BlockDevTest, is_rotational)
   EXPECT_FALSE(sda.is_rotational());
   EXPECT_TRUE(sdb.is_rotational());
 }
+
+TEST(blkdev, _decode_model_enc)
+{
+
+  const char *foo[][2] = {
```
Comment (tchaikov) [inline src/test/common/test_blkdev.cc]: could use an array of pair to utilize the structured binding introduced by C++17, like
```
  const pair<string, string> foo[2] = {
    { "WDC\\x20WDS200T2B0A-00SM50\\x20\\x20\\x20\\x20\\x20\\x20\\x20\\x20\\x20\\x20\\x20\\x20\\x20\\x20\\x20\\x20\\x20\\x20",
      "WDC_WDS200T2B0A-00SM50" },
  };

  for (auto& [model,decoded] : foo) {
    ASSERT_EQ(decoded, _decode_model_enc(model));
  }
```

**Comment by liewegas** [comment]
updated the decode method.. much better now, thanks!  i left hte prints 
because i am frequently frustrated when i hit one of those asserts and 
can't see what the input string was without modifying the code.

**Comment by tchaikov** [comment]
@liewegas probably you forgot to push the updated branch?

**Comment by liewegas** [comment]
retest this please

**Comment by liewegas** [comment]
@tchaikov also updated to keep the python implementation in sync, and to add the sample input + expected output.  (still no tests for those, sadly.)

**Pull request merged**